### PR TITLE
Fix test for flags overriderer, add more tests

### PR DESCRIPTION
**Pull request by matthew-andrews**
@aintgoin2goa

### CODE REVIEW

File: server/fetchFlagJson.js [hunk 1]
```diff
@@ -34,17 +34,17 @@ function initalFetch(opts){
 
 function startPolling(flagsModel, options){
 	var doFetch = function(){
-		superfetch(options.url, {timeout:options.timeout, retry:3})
+		superfetch(options.url, { timeout: options.timeout, retry: 0 })
```
Comment (matthew-andrews) [inline server/fetchFlagJson.js]: i think for the periodic updates we don't need to retry, what do you think mr wilson?

Comment (aintgoin2goa) [inline server/fetchFlagJson.js]: yes, but thinking about it now there may be a bug/feature in superfetch - I think this should be named 'attempts' as I think 0 will mean it never calls it...

Comment (matthew-andrews) [inline server/fetchFlagJson.js]: or how about: `var attempts = (options.retry + 1) || 1;`

Comment (matthew-andrews) [inline server/fetchFlagJson.js]: oops

```js
         var retry = options.retry || 0;
         var attempts = retry + 1;
```

**Comment by aintgoin2goa** [comment]
nice! :+1:

**Pull request merged**